### ISSUE TITLE
Social Meta - OG: & Twitter:description still generated while autogeneration is disabled

### ISSUE THREAD

**Issue by arnaudbroes** [semperfiwebdesign/all-in-one-seo-pack#1656]
Reported here and confirmed by me - https://github.com/semperfiwebdesign/all-in-one-seo-pack/issues/739. I'm guessing that the code for the 'Use Content For Autogenerated OG Descriptions' setting is running regardless or whether or not that setting is activated/deactivated. I'll troubleshoot to see if this was always the case or to see when it broke.

**Comment by artevodesign** [comment]
Hi! Any development on this issue, please?

**Comment by arnaudbroes** [comment]
@artevodesign no updates yet.

**Comment by michaeltorbert** [comment]
@arnaudbroes See when this broke.

**Comment by arnaudbroes** [comment]
@wpsmort the above code (see commit) fixes this issue but the plugin still outputs a OG:description when there is an excerpt (when"Autogenerate descriptions" is disabled). 

Is that the expected behaviour or should there be no OG:description similar to how there is no description unless the user entered one? I'd go with the latter but I'd like your thoughts.

**Comment by wpsmort** [comment]
@arnaudbroes The current behavior is correct.  There's no setting for Autogenerate OG Description in the Social Meta module and it's shouldn't use the Autogenerate Descriptions setting in the General Settings.  Here's why we did it this way:

Social media sites such as Facebook need some basic OG tags in order for a share to be displayed.  Without these necessary tags, when someone shares content, Facebook may return an error and not display the share.  When we developed the Social Meta module, we decided that we'd provide the minimum required OG tags so that sharing would work.  This meant that we had to output an OG:description.  We decided that whatever happened, we'd display this based off of the SEO description, excerpt or content.  

Later on, we added the Use Content For Autogenerated OG Descriptions option to mirror the same option in General Settings.  This was because of WooCommerce and the way they have a Product Short Description (Excerpt) and Product Content and many users wanted to choose which to use for social sharing.  

FYI... I've updated the doc here to make it clearer what we're doing - https://semperplugins.com/documentation/social-meta-settings-individual-pagepost-settings/#description
